jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/nodejs/NodeJsLambdaBuilder.kt

```diff
@@ -56,31 +62,60 @@ class NodeJsLambdaBuilder : LambdaBuilder() {
                         Files.createFile(tsConfig)
                     }
 
-                    // TODO: if there's an existing tsconfig file, should we use it as a base?
-                    tsConfig.writeText(
-                        // language=JSON
-                        """
-                        {
-                            "compilerOptions": {
-                                "${TypeScriptConfig.TYPE_ROOTS}": [
-                                  "${sourceRoot.resolve(TypeScriptConfig.DEFAULT_TYPES_DIRECTORY)}"
-                                ],
-                                "${TypeScriptConfig.TYPES}": [
-                                  "node"
-                                ],
-                                "${TypeScriptConfig.TARGET_OPTION}": "${TypeScriptConfig.LanguageTarget.ES6.libName}",
-                                "${TypeScriptConfig.MODULE}": "${TypeScriptConfig.MODULE_COMMON_JS}",
-                                "${TypeScriptConfig.OUT_DIR}": "$tsOutput",
-                                "${TypeScriptConfig.ROOT_DIR}": ".",
-                                "sourceRoot": "$sourceRoot",
-                                "${TypeScriptConfig.SOURCE_MAP}": true
-                            }
+                    // read base ts config into mutable map
+                    fun loadBaseConfig(tsConfig: Path): MutableMap<String, Any>? {
+                        if (!tsConfig.exists()) {
+                            return null
                         }
-                        """.trimIndent()
-                    )
+
+                        try {
+                            val tsConfigMap: MutableMap<String, Any> = mapper.readValue(tsConfig.reader())
```

Comment:
nit: you can keep as-is, but we've been doing `.inputStream()` elsewhere in the codebase for this pattern